src/interface.jl

```diff
@@ -445,10 +445,20 @@ function ManifoldsBase.get_coordinates!(
     return c
 end
 
+"""
+    point_type_to_vector_type(M::AbstractManifold, TP)
+
+Change point type `TP` that represents point on manifold `M` to matching type for
+representing tangent vectors.
+"""
+point_type_to_vector_type(::AbstractManifold, TP) = TP
```

Comment:
I think until now we did that implicitly, e.g. with `zero_vector(M, p)` we used the type of `p` to generate the right tangent vector type `T`, see for example https://github.com/JuliaManifolds/Manifolds.jl/blob/d55acb407b44298ed077a4edb2d9c875474538fb/src/manifolds/FixedRankMatrices.jl#L838-L853

Reply:
For the name, since the point type is a parameter, we could maybe to /even both)

`tangent_vector_type(M, point_type::Type)`
and its opposite direcection
`point_type(M, vector_type::Type)`
in naming?

The functionality sounds very reasonable, for sure :)

Reply:
(or one could even leave our the `_type`, which leaves the opportunity, that a concrete `tangent_vector(M, p)` could return an allocated `X`.

Reply:
sure `tangent_vector_type` should for now be enough and sure it could be part of `ManifoldsBase`, though then we should be careful to provide that for existing pairs maybe.